diff --git a/ReactCommon/react/renderer/textlayoutmanager/platform/ios/react/renderer/textlayoutmanager/RCTTextLayoutManager.mm b/ReactCommon/react/renderer/textlayoutmanager/platform/ios/react/renderer/textlayoutmanager/RCTTextLayoutManager.mm
--- a/ReactCommon/react/renderer/textlayoutmanager/platform/ios/react/renderer/textlayoutmanager/RCTTextLayoutManager.mm
+++ b/ReactCommon/react/renderer/textlayoutmanager/platform/ios/react/renderer/textlayoutmanager/RCTTextLayoutManager.mm
@@ -386,10 +386,10 @@
     size.height = enumeratedLinesHeight;
   }

-  size = (CGSize){ceil(size.width * layoutContext.pointScaleFactor) / layoutContext.pointScaleFactor,
-                  ceil(size.height * layoutContext.pointScaleFactor) / layoutContext.pointScaleFactor};
+  size = (CGSize){ceil((size.width + 0.001) * layoutContext.pointScaleFactor) / layoutContext.pointScaleFactor,
+                  ceil((size.height + 0.001) * layoutContext.pointScaleFactor) / layoutContext.pointScaleFactor};

   NSRange visibleGlyphRange = [layoutManager glyphRangeForTextContainer:textContainer];

   __block auto attachments = TextMeasurement::Attachments{};

